### DISCUSSION TITLE
2FA is a hassle and not necessarily more secure and should be kept optional

### DISCUSSION THREAD

**Discussion by adorbabble** [community/community#50136] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Hi. This morning, I read an article about how GitHub is going to force everyone to enable 2FA, which I did. But as I don’t want to have to use my phone, give my phone number or install another app like Authy or Microsoft Authenticator, I decided to use TOTP with my password manager, like in this topic : https://github.com/orgs/community/discussions/49715

Now, I haven’t really gained anything from the point of view of security since I don’t use my phone, but there’s an extra step to log in ; it’s as if I had a second password that I keep in my manager…

Furthermore, GitHub wants us to keep some recovery codes as a result, which is another hassle and risk. I’m supposed to store them securely, but if I put them in my password manager, it will just bloat my database, and if I just kept them in a file or wrote them down, I’d risk getting them stolen. I could probably store them in my secure e-mail account, but again I don’t really get any benefit and extra security. For me, the risk of keeping recovery codes and have them stolen is greater than the risk of losing my account because of not keeping them, so I don’t think I will keep them…

I don’t really understand the logic of forcing people to use 2FA

**Comment by enviGit** [comment]
### Hi [adorbabble](https://github.com/adorbabble)!

Well while it may be true that 2FA can be a hassle and it is not necessarily more secure in all cases, it is still a valuable security measure that can prevent many attacks that rely on stolen or compromised passwords. While **it is good to have** the option to choose the type of 2FA that works best for each user, it is important to remember that enforcing 2FA helps protect the **entire** community of users on a platform like GitHub, and can help prevent large-scale security breaches that can have serious consequences. 

While it may seem like an extra hassle to store recovery codes, they serve **an important** purpose in ensuring that you can regain access to your account if you lose your primary 2FA method. Keeping the recovery codes in a secure location, such as a password manager or a safe deposit box, can **greatly** reduce the risk of them being stolen. In contrast, the risk of losing access to your account without a backup method in place could result in the loss of valuable data or even identity theft.

**Comment by MioVisman** [comment]
2FA is complete nonsense, which is invented by "specialists" in security in order to justify their existence.

I haven't used a username/password in years to do anything on this site. How will my password be stolen in this case?
I use Github Desktop to manage my repositories. Will you put 2FA in there too? Or for every commit from Github Desktop I have to go to the site and confirm it via 2FA?

Senseless and ruthless idea. 
Especially if you use the TOTP application on the same computer as the browser.
The Trojan doesn't care where it steals data from.

**Reply by howtophil** [reply]
Agreed. This 2FA bullshit is "security theater" and nothing more. A strong password is more than secure enough.

**Comment by marcinn** [comment]
Time to leave github.

**Comment by ghost** [comment]
yeah, nice to leaving from github!

**Comment by kagaricyan** [comment]
Yeah, Users should not risk the unrecoverable account to maintain the security of the entire community

**Comment by howtophil** [comment]
2FA is theater. It is not superior security. By any measure.

On Mon, Aug 19, 2024 at 4:59 AM kagari ***@***.***> wrote:

> Yeah, Users should not risk the unrecoverable account to maintain the
> security of the entire community
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/community/discussions/50136#discussioncomment-10380850>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMVH3I2HJBGC6WWS54YB2DZSGXVXAVCNFSM6AAAAABMXMKROOVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMZYGA4DKMA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Phillip J Rhoades
HowToPhil.com
KE8GGD
